core/tracker/tracker.go

```diff
@@ -42,11 +46,15 @@ const (
 	sentinel
 )
 
+// shareIdx represents a peer's share index.
+type shareIdx int
+
 // event represents an event emitted by a core workflow component.
 type event struct {
 	duty      core.Duty
 	component component
 	pubkey    core.PubKey
+	shareIdx  shareIdx
```

Comment:
make a note that this is optional field only set for some component events and list them.